src/coprocessor/dag/expr/json.rs

```diff
@@ -57,11 +57,11 @@ impl FnCall {
         row: &'a [Datum],
     ) -> Result<Option<Cow<'a, Json>>> {
         let mut pairs = BTreeMap::new();
-        if !self.children.is_empty() {
-            let parser = JsonFuncArgsParser::new(ctx, row);
-            let keys = try_opt!(parser.get_strings(self.children.iter().step_by(2)));
-            let elems = try_opt!(parser.get_jsons(self.children[1..].iter().step_by(2)));
-            pairs.extend(keys.into_iter().zip(elems.into_iter()));
+        let parser = JsonFuncArgsParser::new(ctx, row);
+        for chunk in self.children.chunks(2) {
+            let key = try_opt!(chunk[0].eval_string_and_decode(ctx, row)).into_owned();
+            let val = try_opt!(parser.get_json(&chunk[1]));
+            pairs.insert(key, val);
```

Comment:
Could we check None before any `insert`?

Reply:
How?